sahi/model.py

```diff
@@ -663,3 +664,167 @@ def _create_object_prediction_list_from_original_predictions(
+        from sahi.utils.torchvision import COCO_CLASSES
+
+        if self.category_mapping is None:
+            category_names = {str(i): COCO_CLASSES[i] for i in range(len(COCO_CLASSES))}
+            self.category_mapping = category_names
+
+    def perform_inference(self, image: np.ndarray, image_size: int = None):
+        """
+        Prediction is performed using self.model and the prediction result is set to self._original_predictions.
+        Args:
+            image: np.ndarray
+                A numpy array that contains the image to be predicted. 3 channel image should be in RGB order.
+            image_size: int
+                Inference input size.
+        """
+        try:
+            import torchvision
+        except ImportError:
+            raise ImportError("torchvision is not installed. Please install torchvision to use this model.")
+
+        # Confirm model is loaded
+        assert self.model is not None, "Model is not loaded, load it by calling .load_model()"
+
+        from sahi.utils.torchvision import numpy_to_torch, resize_image
+
+        if self.image_size is not None:
+            image = resize_image(image, self.image_size)
+            image = numpy_to_torch(image)
+            prediction_result = self.model([image])
+
+        else:
+            prediction_result = self.model([image])
+
+        self._original_predictions = prediction_result
+
+    @property
+    def num_categories(self):
+        """
+        Returns number of categories
+        """
+        return len(self.category_mapping)
+
+    @property
+    def has_mask(self):
+        """
+        Returns if model output contains segmentation mask
+        """
+        has_mask = self.model.with_mask
+        return has_mask
+
+    @property
+    def category_names(self):
+        return self.category_mapping
+
+    def _create_object_prediction_list_from_original_predictions(
+        self,
+        shift_amount_list: Optional[List[List[int]]] = [[0, 0]],
+        full_shape_list: Optional[List[List[int]]] = None,
+    ):
+        """
+        self._original_predictions is converted to a list of prediction.ObjectPrediction and set to
+        self._object_prediction_list_per_image.
+        Args:
+            shift_amount_list: list of list
+                To shift the box and mask predictions from sliced image to full sized image, should
+                be in the form of List[[shift_x, shift_y],[shift_x, shift_y],...]
+            full_shape_list: list of list
+                Size of the full image after shifting, should be in the form of
+                List[[height, width],[height, width],...]
+        """
+        original_predictions = self._original_predictions
+        category_mapping = self.category_mapping
+
+        # compatilibty for sahi v0.8.15
+        if isinstance(shift_amount_list[0], int):
+            shift_amount_list = [shift_amount_list]
+        if full_shape_list is not None and isinstance(full_shape_list[0], int):
```

Comment:
Bu if bloğundaki expression biraz tehlikeli çünkü None olmadığı durumda Iterable olmasının garanti olması bekleniyor, olmazsa istenmeyen exceptionlara yol açar (indexerror, vb.)

```python
if isinstance(full_shape_list, list) and isinstance(full_shape_list[0], int):
```
burada `isinstance()`'taki type tabi `full_shape_list`'in beklenen type'ı olmalı (list olarak varsaydım, eğer değilse ona göre değiştirelim.)

Reply:
@devrimcavusoglu `full_shape_list`in liste olmasi zorunlu, list seklinde gelmiyorsa hata vermesi sorun olmaz diye dusunuyorum